v2/internal/controllers/controller_resources.go

```diff
@@ -6,16 +6,33 @@ Licensed under the MIT license.
 package controllers
 
 import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1alpha1api20200601"
+	. "github.com/Azure/azure-service-operator/v2/internal/logging"
+	"github.com/Azure/azure-service-operator/v2/internal/reflecthelpers"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/registration"
 )
 
-func GetKnownStorageTypes() []client.Object {
+func GetKnownStorageTypes() []registration.StorageTypeInfo {
```

Comment:
Since this is in the `registration` package, consider renamign this to `registration.StorageType`.

Reply:
Dropped the `Registration` suffix from the other two types too, as they were repeating themselves (`registration.WatchRegistration`)